fla/modules/convolution.py

```
@@ -789,16 +789,16 @@ def forward(
-            ((cu_seqlens is not None or seq_idx is not None) and cache is not None) or \
-                (cu_seqlens is not None and output_final_state):
-            warnings.warn(
-                "The CUDA backend does not support both `cu_seqlens` and `cache` being provided, "
-                "or both `cu_seqlens` and `output_final_state` being provided. "
-                "Switching to the Triton backend instead. "
-            )
-            self.backend = 'triton'
+        # 2. both `cu_seqlens` and `output_final_state` being provided
+        if self.backend == 'cuda':
+            if ((cu_seqlens is not None or seq_idx is not None) and cache is not None) or \
+                    (cu_seqlens is not None and output_final_state):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)